rosmon_core/src/main.cpp

```diff
@@ -370,29 +370,38 @@ int main(int argc, char** argv)
 				logDir.pop_back();
 			}
 			workDir = logDir + "/rosmon";
-			if (chdir(workDir.c_str()) == 0 || mkdir(workDir.c_str(), 0777) == 0) 
-			{
-				logDir = workDir + "/core_dumps";
-				if (!(chdir(logDir.c_str()) == 0 || mkdir(logDir.c_str(), 0777) == 0))
+			try {
+				if (fs::exists(workDir) || mkdir(workDir.c_str(), 0777) == 0) 
```

Comment:
It would be more consistent, remove nesting and improve readability if this type of check was always
```
if (!(fs::exists(workDir) || mkdir(workDir.c_str(), 0777) == 0))  {
    fmtNoThrow::print(stderr, "Could not create rosmon directory\n");
￼    return 1;
}
```
Personally, I also find `if (!fs::exists(workDir) && mkdir(workDir.c_str(), 0777) != 0)` easier to digest